### ISSUE TITLE
encrypt_variable's `delete_GitHub_token` after raise_for_status

### ISSUE THREAD

**Issue by Carreau** [drdoctr/doctr#174]
I'm guessing we want it before.

Should it be in a context manager maybe ?

**Comment by asmeurer** [comment]
It's only used one place, so a try/finally should be fine. I don't have any private repos to test against, so cautious to fiddle with the private repo code too much.

**Comment by Carreau** [comment]
> It's only used one place, so a try/finally should be fine. I don't have any private repos to test against, so cautious to fiddle with the private repo code too much.

Yes, for sure. I'll just need this code for non-private repo which need authentication with travis. 
There is (BTW) a conflation private with need authentication but not relevant to this issue.

**Comment by asmeurer** [comment]
I had wondered about that too, but according to @dan-blanchard at https://github.com/drdoctr/doctr/pull/121#issuecomment-248379855 you can only use travis-ci.com with private GitHub repos (or are you referring to something different?).

**Comment by Carreau** [comment]
Nah, i'm referring to #166 , where setting an encrypted variable on a public repo needs a similar piece of code but with `.org` so where authentication and `.org`/`.com` are unrelated concepts. 

It's still also sufficiently different that it may not be worth to consolidate into an abstracted function.
